src/main/kotlin/sliv/tool/scene/model/Layer.kt

```diff
@@ -1,13 +1,29 @@
 package sliv.tool.scene.model
 
+import javafx.scene.paint.Color
+
 //Stores common context for landmarks drawing.
 //Layers properties being edited in the settings menu.
 //Settings menu appearance depends on type of the corresponding layer.
 //Meaningful changes here provokes scene redrawing.
 sealed class Layer(val name: String) {
-    class PointLayer(name: String) : Layer(name)
-    class LineLayer(name: String) : Layer(name)
+    class PointLayer(name: String) : Layer(name) {
+        var color: Color = DEFAULT_COLOR
+    }
+
+    class LineLayer(name: String) : Layer(name) {
+        private val colorMap = HashMap<Long, Color>()
+        fun getColor(landmark: Landmark): Color {
```

Comment:
why do layer evaluate its color?

Reply:
Layer also evaluate opacity, width, thickness, visibility and etc. It is very appropriate to evaluate color alongside. For keypoints the color is directly set from the property in the layer. There is no gains to use something additional.